src/lm/arpa-lm-compiler.h

```diff
@@ -31,7 +31,7 @@ class ArpaLmCompilerImplInterface;
 
 class ArpaLmCompiler : public ArpaFileParser {
  public:
-  ArpaLmCompiler(ArpaParseOptions options, int sub_eps,
+  ArpaLmCompiler(const ArpaParseOptions& options, int sub_eps,
```

Comment:
I'm not sure about the intended effect of changes here (and similar ones elsewhere) for making the variables `const` and pointers. I hope you have got them approved by @danpovey.

Reply:
& ---> to avoid unnecessary copy
const ---> we do not need to change it inside the constructor.

Reply:
Ah, I see, a pass by reference for the arguments which will not be changes inside the function.